core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -0,0 +1,381 @@
+    'topics_and_skills_dashboard', 'topic_editor', 'skill_editor',
+    'story_editor'];
+  currentWindowWidth = this.windowDimensionsService.getWidth();
+  // The order of the elements in this array specifies the order in
+  // which they will be hidden. Earlier elements will be hidden first.
+  NAV_ELEMENTS_ORDER = [
+    'I18N_TOPNAV_DONATE', 'I18N_TOPNAV_CLASSROOM', 'I18N_TOPNAV_ABOUT',
+    'I18N_CREATE_EXPLORATION_CREATE', 'I18N_TOPNAV_LIBRARY'];
+
+  CLASSROOM_PROMOS_ARE_ENABLED = false;
+  googleSignInIconUrl = this.urlInterpolationService.getStaticImageUrl(
+    '/google_signin_buttons/google_signin.svg');
+  navElementsVisibilityStatus ={};
+
+  constructor(
+    private classroomBackendApiService: ClassroomBackendApiService,
+    private sidebarStatusService: SidebarStatusService,
+    private urlInterpolationService: UrlInterpolationService,
+    private debouncerService: DebouncerService,
+    private navigationService: NavigationService,
+    private siteAnalyticsService: SiteAnalyticsService,
+    private userService: UserService,
+    private deviceInfoService: DeviceInfoService,
+    private windowDimensionsService: WindowDimensionsService,
+    private searchService: SearchService,
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private windowRef: WindowRef,
+    private userBackendApiService: UserBackendApiService,
+    private focusManagerService: FocusManagerService
+  ) {}
+
+  ngOnInit(): void {
+    this.getProfileImageDataAsync();
+    this.currentUrl = window.location.pathname.split('/')[1];
+    this.labelForClearingFocus = AppConstants.LABEL_FOR_CLEARING_FOCUS;
+    this.focusManagerService.setFocus(this.labelForClearingFocus);
+    this.logoutUrl = AppConstants.LOGOUT_URL;
+    this.userMenuIsShown = (this.currentUrl !== this.NAV_MODE_SIGNUP);
+    this.inClassroomPage = false;
+    this.standardNavIsShown = (
+      this.NAV_MODES_WITH_CUSTOM_LOCAL_NAV.indexOf(this.currentUrl) === -1);
+    if (this.currentUrl === 'learn') {
+      this.inClassroomPage = true;
+    }
+    this.ACTION_OPEN = this.navigationService.ACTION_OPEN;
+    this.ACTION_CLOSE = this.navigationService.ACTION_CLOSE;
+    this.KEYBOARD_EVENT_TO_KEY_CODES =
+      this.navigationService.KEYBOARD_EVENT_TO_KEY_CODES;
+    this.windowIsNarrow = this.windowDimensionsService.isWindowNarrow();
+
+    let service = this.classroomBackendApiService;
+    service.fetchClassroomPromosAreEnabledStatusAsync().then(
+      (classroomPromosAreEnabled) => {
+        this.CLASSROOM_PROMOS_ARE_ENABLED = classroomPromosAreEnabled;
+      });
+
+    this.directiveSubscriptions.add(
+      this.searchService.onSearchBarLoaded.subscribe(
+        () => {
+          setTimeout(this.truncateNavbar, 100);
+        }
+      )
+    );
+
+    this.userService.getUserInfoAsync().then((userInfo) => {
+      if (userInfo.getPreferredSiteLanguageCode()) {
+        // This.translate.use(userInfo.getPreferredSiteLanguageCode());
```

Comment:
No commented out code please

Reply:
We don't need to invoke `$translate.use()` from here, now.
It will now automatically pick it up from i18nLanguageCodeService.
@mridul-netizen  Can you please remove this commented line?.

Reply:
done